File: Практика/Задание10/HelloParallel.java
package ip.ivanov.danil.Практика.Задание10;

public class HelloParallel {

    public static void main(String[] args) {
        // Запуск 5 параллельных потоков
        for (int i = 1; i <= 5; i++) {
            int processNumber = i;
            new Thread(() -> printHello(processNumber)).start();
        }
    }

    // Функция для вывода сообщения с задержкой
    private static void printHello(int processNumber) {
        for (int i = 1; i <= 5; i++) {
            try {
                // Выводим сообщение
                System.out.println("Hello from " + processNumber + "!");
                // Задержка 1 секунда
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
        }
    }
}

File: Практика/Задание9/hello_parallel.erl
-module(hello_parallel).
        -export([start/0]).

        start() ->
        % Запускаем 5 параллельных процессов
        lists:foreach(fun(N) -> spawn(fun() -> print_hello(N) end) end, lists:seq(1, 5)).

        % Функция для вывода сообщения с задержкой
        print_hello(N) ->
        lists:foreach(fun(_) ->
        io:format("Hello from ~p!~n", [N]),
        timer:sleep(1000) % Задержка 1 секунда
        end, lists:seq(1, 5)).


        hello_parallel:start().
